dcmgr/lib/dcmgr/drivers/hypervisor/linux_hypervisor.rb

```diff
@@ -59,6 +59,8 @@ def check_interface(hc)
           next if vnic[:network].nil?
 
           network = hc.rpc.request('hva-collector', 'get_network', vnic[:network_id])
+          
+          raise "Dc network missing in database." if network[:dc_network][:name].nil?              
```

Comment:
You are still checking the wrong value. If `network[:dc_network]` returns nil, it will crash because you're calling the `[]` method on a nil object. Besides, it's impossible for name to return nil as the name column in the dc_networks table doesn't allow null values.